src/utils/apiUtils.ts

```diff
@@ -3,10 +3,14 @@ import { apiUrl } from './constants'
 export const fetchData = async (
   endpoint: string,
   config?: RequestInit
-): Promise<Response> => {
-  const request = await fetch(`${apiUrl}/${endpoint}`, config)
-  const response = request.json()
-  return response
+): Promise<Response | undefined> => {
+  try {
+    const request = await fetch(`${apiUrl}/${endpoint}`, config)
+    const response = request.json()
+    return response
+  } catch (err) {
+    console.log(err)
```

Comment:
any suggestions for better error handling?

Reply:
check out `fetchText` in `utils`